src/main/java/com/o19s/es/ltr/stats/suppliers/StoreStatsSupplier.java

```diff
@@ -0,0 +1,138 @@
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.aggregations.AggregationBuilders;
+import org.elasticsearch.search.aggregations.bucket.terms.Terms;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * A supplier which provides information on all feature stores. It provides basic
+ * information such as the index health and count of feature sets, features and
+ * models in the store.
+ */
+public class StoreStatsSupplier implements Supplier<Map<String, Map<String, Object>>> {
+    private static final Logger LOG = LogManager.getLogger(StoreStatsSupplier.class);
+    private static final String AGG_FIELD = "type";
+    private final Client client;
+    private final ClusterService clusterService;
+    private final IndexNameExpressionResolver indexNameExpressionResolver;
+
+    public StoreStatsSupplier(Client client, ClusterService clusterService) {
+        this.client = client;
+        this.clusterService = clusterService;
+        this.indexNameExpressionResolver = new IndexNameExpressionResolver();
+    }
+
+    @Override
+    public Map<String, Map<String, Object>> get() {
+        String[] names = indexNameExpressionResolver.concreteIndexNames(clusterService.state(),
+                new ClusterStateRequest().indices(
+                        IndexFeatureStore.DEFAULT_STORE, IndexFeatureStore.STORE_PREFIX + "*"));
+        final MultiSearchRequestBuilder requestBuilder = client.prepareMultiSearch();
+        List<String> indices = new ArrayList<>();
+        Stream.of(names)
+                .filter(IndexFeatureStore::isIndexStore)
+                .map(s -> clusterService.state().metadata().index(s))
+                .filter(Objects::nonNull)
+                .map(IndexMetadata::getIndex)
+                .map(Index::getName)
+                .forEach(idx -> {
+                    indices.add(idx);
+                    requestBuilder.add(countSearchRequest(idx));
+                });
+        return createStoreStatsResponse(requestBuilder, indices);
+    }
+
+    private Map<String, Map<String, Object>> createStoreStatsResponse(MultiSearchRequestBuilder requestBuilder,
+                                                                      List<String> indices) {
+        try {
+            MultiSearchResponse msr = requestBuilder.execute().get();
+            assert indices.size() == msr.getResponses().length;
+            Map<String, Map<String, Object>> stats = new ConcurrentHashMap<>(indices.size());
```

Comment:
I don't think you need a concurrent hashmap here, if you need one I probably missed something in your PR.